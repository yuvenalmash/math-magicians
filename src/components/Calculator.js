import React from 'react';
import './calculator.css';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.handleCalculate = this.handleCalculate.bind(this);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

  handleCalculate(e) {
    e.preventDefault();
    const buttonName = e.target.name;
    const newState = calculate(this.state, buttonName);
    this.setState(newState);
  }

  render() {
    const { total, operation, next } = this.state;
    return (
      <main>
        <div id="itemInput">
          {total}
          {operation}
          {next}
        </div>
        <button id="item0" className="btn" name="0" onClick={this.handleCalculate} type="button">0</button>
        <button id="item1" className="btn" name="1" onClick={this.handleCalculate} type="button">1</button>
        <button id="item2" className="btn" name="2" onClick={this.handleCalculate} type="button">2</button>
        <button id="item3" className="btn" name="3" onClick={this.handleCalculate} type="button">3</button>
        <button id="item4" className="btn" name="4" onClick={this.handleCalculate} type="button">4</button>
        <button id="item5" className="btn" name="5" onClick={this.handleCalculate} type="button">5</button>
        <button id="item6" className="btn" name="6" onClick={this.handleCalculate} type="button">6</button>
        <button id="item7" className="btn" name="7" onClick={this.handleCalculate} type="button">7</button>
        <button id="item8" className="btn" name="8" onClick={this.handleCalculate} type="button">8</button>
        <button id="item9" className="btn" name="9" onClick={this.handleCalculate} type="button">9</button>
        <button id="itemAC" className="btn" name="AC" onClick={this.handleCalculate} type="button">AC</button>
        <button id="itemPlusMinus" className="btn" name="+/-" onClick={this.handleCalculate} type="button">+/-</button>
        <button id="itemModulus" className="btn" name="%" onClick={this.handleCalculate} type="button">%</button>
        <button id="itemDivide" className="btn special" name="÷" onClick={this.handleCalculate} type="button">÷</button>
        <button id="itemMultiply" className="btn special" name="x" onClick={this.handleCalculate} type="button">×</button>
        <button id="itemMinus" className="btn special" name="-" onClick={this.handleCalculate} type="button">-</button>
        <button id="itemPlus" className="btn special" name="+" onClick={this.handleCalculate} type="button">+</button>
        <button id="itemEqual" className="btn special" name="=" onClick={this.handleCalculate} type="button">=</button>
        <button id="itemPeriod" className="btn" name="." onClick={this.handleCalculate} type="button">•</button>
      </main>
    );
  }
}

export default Calculator;
