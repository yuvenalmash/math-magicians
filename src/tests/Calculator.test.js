import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('check the rendered matches the snapshot', () => {
  const calculator = renderer
    .create(<Calculator />)
    .toJSON();
  expect(calculator).toMatchSnapshot();
});
