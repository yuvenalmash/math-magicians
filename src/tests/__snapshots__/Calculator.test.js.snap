// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the rendered matches the snapshot 1`] = `
<div
  className="calcContainer"
>
  <div
    id="itemInput"
  >
    0
  </div>
  <button
    className="btn"
    id="item0"
    name="0"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="btn"
    id="item1"
    name="1"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="btn"
    id="item2"
    name="2"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="btn"
    id="item3"
    name="3"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="btn"
    id="item4"
    name="4"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="btn"
    id="item5"
    name="5"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="btn"
    id="item6"
    name="6"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="btn"
    id="item7"
    name="7"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="btn"
    id="item8"
    name="8"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="btn"
    id="item9"
    name="9"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="btn"
    id="itemAC"
    name="AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="btn"
    id="itemPlusMinus"
    name="+/-"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="btn"
    id="itemModulus"
    name="%"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="btn special"
    id="itemDivide"
    name="÷"
    onClick={[Function]}
    type="button"
  >
    ÷
  </button>
  <button
    className="btn special"
    id="itemMultiply"
    name="x"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
  <button
    className="btn special"
    id="itemMinus"
    name="-"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="btn special"
    id="itemPlus"
    name="+"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="btn special"
    id="itemEqual"
    name="="
    onClick={[Function]}
    type="button"
  >
    =
  </button>
  <button
    className="btn"
    id="itemPeriod"
    name="."
    onClick={[Function]}
    type="button"
  >
    •
  </button>
</div>
`;
